lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -0,0 +1,760 @@
+    acme_challenge_path:
+        required: false
+        description:
+            - Path to the ACME challenge directory that is served on http://<HOST>:80/.well-known/acme-challenge/
+
+    signature_algorithms:
+        required: false
+        description:
+            - list of algorithms that you would accept the certificate to be signed with
+              (e.g. ['sha256WithRSAEncryption', 'sha512WithRSAEncryption']).
+
+    issuer:
+        required: false
+        description:
+            - Key/value pairs that must be present in the issuer name field of the certificate
+
+    subject:
+        required: false
+        description:
+            - Key/value pairs that must be present in the subject name field of the certificate
+
+    has_expired:
+        required: false
+        choices: [ True, False ]
+        description:
+            - Checks if the certificate is expired/not expired at the time the module is executed.
+
+    version:
+        required: false
+        description:
+            - Version of the certificate. Nowadays it should almost always be 3.
+
+    valid_at:
+        required: false
+        description:
+            - The certificate must be valid at this point in time. The timestamp is formatted as an ASN.1 TIME.
+
+    invalid_at:
+        required: false
+        description:
+            - The certificate must be invalid at this point in time. The timestamp is formatted as an ASN.1 TIME.
+
+    notBefore:
+        required: false
+        description:
+            - The certificate must start to become valid at this point in time. The timestamp is formatted as an ASN.1 TIME.
+
+    notAfter:
+        required: false
+        description:
+            - The certificate must expire at this point in time. The timestamp is formatted as an ASN.1 TIME.
+
+    valid_in:
+        required: false
+        description:
+            - The certificate must still be valid in `valid_in` seconds from now.
+
+    keyUsage:
+        required: false
+        description:
+            - The keyUsage extension field must contain all these values.
+    keyUsage_strict:
+        required: false
+        description:
+            - If set to True, The keyUsage extension field must contain only these values.
+
+    extendedKeyUsage:
+        required: false
+        description:
+            - The extendedKeyUsage extension field must contain all these values.
+
+    extendedKeyUsage_strict:
+        required: false
+        description:
+            - If set to True, The extendedKeyUsage extension field must contain only these values.
+
+    subjectAltName:
+        required: false
+        description:
+            - The subjectAltName extension field must contain these values.
+
+    subjectAltName_strict:
+        required: false
+        description:
+            - If set to True, The subjectAltName extension field must contain only these values.
+'''
+
+
+EXAMPLES = '''
+- name: Generate a Self Signed OpenSSL certificate
```

Comment:
This one will fail, since selfsigned_notBefore and selfsigned_notAfter have changed their meaning from the initial draft and don't have default values any more.